From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <28825609+KennyTV@users.noreply.github.com>
Date: Sun, 15 Dec 2019 13:33:44 +0100
Subject: [PATCH] Player database id


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 97954bf5e7c0eb243d82902baa88bab030c25995..239569fb971afafaa067c749d6b367dc0b4cab86 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -113,6 +113,11 @@ public final class Bukkit {
     public static net.luminu.paper.locale.PaperLocaleProvider getLocaleProvider() {
         return server.getLocaleProvider();
     }
+
+    @Nullable
+    public static Player getPlayer(int databaseId) {
+        return server.getPlayer(databaseId);
+    }
     // Luminu end
 
     /**
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index f9b3a7316f84b917fe02444419055770bd02ecfe..57fcc76b43d74e8416f33f0d5a35dc08645f9be4 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -93,6 +93,9 @@ public interface Server extends PluginMessageRecipient {
 
     @Nullable // only during initialization until it's set
     public net.luminu.paper.locale.PaperLocaleProvider getLocaleProvider();
+
+    @Nullable
+    public Player getPlayer(int databaseId);
     // Luminu end
 
     /**
diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index bdeb64c5c94d5f37fa437e527c37bea58d32aeec..01fc92b263dfdabb6801de4752fcb3a582ebde23 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -652,6 +652,23 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     public void sendActionBar(char alternateChar, @NotNull String message);
 
     // Luminu start
+    /**
+     * Returns the unique user id of the player.
+     *
+     * @return luminu interal, unique user id
+     */
+    default int getDatabaseId() {
+        return 0;
+    }
+
+    /**
+     * Sets the internal, unique database user id.
+     *
+     * @param databaseId database id
+     */
+    default void setDatabaseId(int databaseId) {
+    }
+
     /**
      * Sends an Action Bar message to the client.
      *
