From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <28825609+KennyTV@users.noreply.github.com>
Date: Sun, 15 Dec 2019 13:33:44 +0100
Subject: [PATCH] Player database id


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index ffe0ebd9866a7899c1925cd4bfe22022236b2b8c..f73339de13bbdb72e1f7d106944042652412a5fd 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -113,6 +113,11 @@ public final class Bukkit {
     public static net.luminu.paper.locale.PaperLocaleProvider getLocaleProvider() {
         return server.getLocaleProvider();
     }
+
+    @Nullable
+    public static Player getPlayer(int databaseId) {
+        return server.getPlayer(databaseId);
+    }
     // Luminu end
 
     /**
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 90ed0eb6fc9cd914f8e7707472bd16d2f5ec3bf4..d8ca34912b70c760ca0a09627f30d297576af6ee 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -93,6 +93,9 @@ public interface Server extends PluginMessageRecipient {
 
     @Nullable // only during initialization until it's set
     public net.luminu.paper.locale.PaperLocaleProvider getLocaleProvider();
+
+    @Nullable
+    public Player getPlayer(int databaseId);
     // Luminu end
 
     /**
diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 0c3f44e647204b4795176011f784fcf0c2d14a92..42e2fab033a8e708599b7ab888fa8f5e3de1da94 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -666,6 +666,23 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     public void sendActionBar(char alternateChar, @NotNull String message);
 
     // Luminu start
+    /**
+     * Returns the unique user id of the player.
+     *
+     * @return luminu interal, unique user id
+     */
+    default int getDatabaseId() {
+        return 0;
+    }
+
+    /**
+     * Sets the internal, unique database user id.
+     *
+     * @param databaseId database id
+     */
+    default void setDatabaseId(int databaseId) {
+    }
+
     /**
      * Sends an Action Bar message to the client.
      *
