From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <kennytv@t-online.de>
Date: Sat, 1 Feb 2020 17:36:31 +0100
Subject: [PATCH] Suppress absolutely unnecessary classloading warning


diff --git a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
index 2abd140ce238c511c513179e360782ab7ecb53c8..fb5dca2ad5bc85747feb6c6865273c6b5103312f 100644
--- a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
+++ b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
@@ -115,6 +115,7 @@ public final class PluginClassLoader extends URLClassLoader { // Spigot
                 result = loader.getClassByName(name);
 
                 if (result != null) {
+                    if (true) return result; // Lumi - just fuck off, please
                     PluginDescriptionFile provider = ((PluginClassLoader) result.getClassLoader()).description;
 
                     if (provider != description
