From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <28825609+KennyTV@users.noreply.github.com>
Date: Wed, 31 Jul 2019 13:23:20 +0200
Subject: [PATCH] Readd servername property


diff --git a/src/main/java/net/minecraft/server/DedicatedServerProperties.java b/src/main/java/net/minecraft/server/DedicatedServerProperties.java
index 1122861a8f9c12668e7bd9d65e63fc9166a66d1a..410d2595258cbf13900877ca4dd9d2872cb85eb0 100644
--- a/src/main/java/net/minecraft/server/DedicatedServerProperties.java
+++ b/src/main/java/net/minecraft/server/DedicatedServerProperties.java
@@ -54,6 +54,7 @@ public class DedicatedServerProperties extends PropertyManager<DedicatedServerPr
     public final PropertyManager<DedicatedServerProperties>.EditableProperty<Integer> playerIdleTimeout;
     public final PropertyManager<DedicatedServerProperties>.EditableProperty<Boolean> whiteList;
     public final GeneratorSettings generatorSettings;
+    public String serverName; // Luminu - readd (configurable) servername
 
     public final String rconIp; // Paper - Add rcon ip
 
@@ -111,6 +112,7 @@ public class DedicatedServerProperties extends PropertyManager<DedicatedServerPr
         final String rconIp = this.getSettingIfExists("rcon.ip");
         this.rconIp = rconIp == null ? this.serverIp : rconIp;
         // Paper end
+        this.serverName = this.getString("server-name", "Unknown server"); // Luminu - readd servername
     }
 
     // CraftBukkit start
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 94005471723187cba438670a7e52363959ad4d02..98be7bec56844c65832b6e20d9613cd16d6493ac 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -504,6 +504,13 @@ public final class CraftServer implements Server {
         return serverName;
     }
 
+    // Luminu start - readd servername
+    @Override
+    public String getServerName() {
+        return getProperties().serverName;
+    }
+    // Luminu end
+
     @Override
     public String getVersion() {
         return serverVersion + " (MC: " + console.getVersion() + ")";
