From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <28825609+KennyTV@users.noreply.github.com>
Date: Sat, 3 Aug 2019 21:17:12 +0200
Subject: [PATCH] Accessibility changes


diff --git a/src/main/java/net/minecraft/server/PacketPlayOutPlayerInfo.java b/src/main/java/net/minecraft/server/PacketPlayOutPlayerInfo.java
index 9a48a4e90c1b1cbcc8dc604fdf0f20a6d0f3bd22..ca0686320f4fe00d0b72fbeaf55f0cc80df9734f 100644
--- a/src/main/java/net/minecraft/server/PacketPlayOutPlayerInfo.java
+++ b/src/main/java/net/minecraft/server/PacketPlayOutPlayerInfo.java
@@ -12,7 +12,7 @@ import javax.annotation.Nullable;
 public class PacketPlayOutPlayerInfo implements Packet<PacketListenerPlayOut> {
 
     private PacketPlayOutPlayerInfo.EnumPlayerInfoAction a;
-    private final List<PacketPlayOutPlayerInfo.PlayerInfoData> b = Lists.newArrayList();
+    public final List<PacketPlayOutPlayerInfo.PlayerInfoData> b = Lists.newArrayList(); // Luminu - make public
 
     public PacketPlayOutPlayerInfo() {}
 
diff --git a/src/main/java/net/minecraft/server/TileEntitySkull.java b/src/main/java/net/minecraft/server/TileEntitySkull.java
index fb7aa15c272678be7ec2a1d28891a62121695ed8..5c7110cd7843400574eef9e98ad9754e45e4ec6c 100644
--- a/src/main/java/net/minecraft/server/TileEntitySkull.java
+++ b/src/main/java/net/minecraft/server/TileEntitySkull.java
@@ -40,7 +40,7 @@ public class TileEntitySkull extends TileEntity /*implements ITickable*/ { // Pa
                     .setNameFormat("Head Conversion Thread - %1$d")
                     .build()
     );
-    public static final LoadingCache<String, GameProfile> skinCache = CacheBuilder.newBuilder()
+    public static LoadingCache<String, GameProfile> skinCache = CacheBuilder.newBuilder() // Luminu - remove final modifier
             .maximumSize( 5000 )
             .expireAfterAccess( 60, TimeUnit.MINUTES )
             .build( new CacheLoader<String, GameProfile>()
