From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <28825609+KennyTV@users.noreply.github.com>
Date: Sat, 3 Aug 2019 21:27:09 +0200
Subject: [PATCH] Localized messages


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 98be7bec56844c65832b6e20d9613cd16d6493ac..3e819b6d7028f279236f3bd3d4a00b60f0a3d16e 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -505,10 +505,17 @@ public final class CraftServer implements Server {
     }
 
     // Luminu start - readd servername
+    public net.luminu.paper.locale.PaperLocaleProvider localeProvider;
+
     @Override
     public String getServerName() {
         return getProperties().serverName;
     }
+
+    @Override
+    public net.luminu.paper.locale.PaperLocaleProvider getLocaleProvider() {
+        return localeProvider;
+    }
     // Luminu end
 
     @Override
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 1b84d4f26da310c94510fa4c0da9a68ec8c6b71c..f72730cade1297dc8559638ddebe41c9af0e71ff 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1927,6 +1927,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     }
 
     @Override
+    @Deprecated // Luminu - deprecate
     public String getLocale() {
         // Paper start - Locale change event
         final String locale = getHandle().locale;
@@ -1934,6 +1935,20 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         // Paper end
     }
 
+    // Luminu start
+    private java.util.Locale locale = java.util.Locale.GERMAN;
+
+    @Override
+    public java.util.Locale getSetLocale() {
+        return locale;
+    }
+
+    @Override
+    public void setLocale(java.util.Locale locale) {
+        this.locale = locale;
+    }
+    // Luminu end
+
     // Paper start
     public void setAffectsSpawning(boolean affects) {
         this.getHandle().affectsSpawning = affects;
