From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <28825609+KennyTV@users.noreply.github.com>
Date: Sat, 3 Aug 2019 21:27:09 +0200
Subject: [PATCH] Localized messages


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 5ee80bb8047d466dc363747bb348282e19b35c69..174c8215d507ab7dc09637545354dc1b67729050 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -509,10 +509,17 @@ public final class CraftServer implements Server {
     }
 
     // Luminu start - readd servername
+    public net.luminu.paper.locale.PaperLocaleProvider localeProvider;
+
     @Override
     public String getServerName() {
         return getProperties().serverName;
     }
+
+    @Override
+    public net.luminu.paper.locale.PaperLocaleProvider getLocaleProvider() {
+        return localeProvider;
+    }
     // Luminu end
 
     @Override
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 72cd52fd7f0ba8df5be8989111d76e7180f033d7..723996a703efea72d3dbf7e84a0555891b3106a8 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1959,6 +1959,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     }
 
     @Override
+    @Deprecated // Luminu - deprecate
     public String getLocale() {
         // Paper start - Locale change event
         final String locale = getHandle().locale;
@@ -1966,6 +1967,20 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         // Paper end
     }
 
+    // Luminu start
+    private java.util.Locale locale = java.util.Locale.GERMAN;
+
+    @Override
+    public java.util.Locale getSetLocale() {
+        return locale;
+    }
+
+    @Override
+    public void setLocale(java.util.Locale locale) {
+        this.locale = locale;
+    }
+    // Luminu end
+
     // Paper start
     public void setAffectsSpawning(boolean affects) {
         this.getHandle().affectsSpawning = affects;
