From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <28825609+KennyTV@users.noreply.github.com>
Date: Sat, 3 Aug 2019 21:27:09 +0200
Subject: [PATCH] Localized messages


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 16bb42f213d362eb03442282df49b403bfd7714f..68d852621d580c44a5e6c13f93daf4a9b0065dc0 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -479,10 +479,17 @@ public final class CraftServer implements Server {
     }
 
     // Luminu start - readd servername
+    public net.luminu.paper.locale.PaperLocaleProvider localeProvider;
+
     @Override
     public String getServerName() {
         return getProperties().serverName;
     }
+
+    @Override
+    public net.luminu.paper.locale.PaperLocaleProvider getLocaleProvider() {
+        return localeProvider;
+    }
     // Luminu end
 
     @Override
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 33cac5fcb1b8bfeaba38c2b537f22349de3e832f..569d1a535f3ac05849af907d78484a81298193db 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1876,6 +1876,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     }
 
     @Override
+    @Deprecated // Luminu - deprecate
     public String getLocale() {
         // Paper start - Locale change event
         final String locale = getHandle().locale;
@@ -1883,6 +1884,20 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         // Paper end
     }
 
+    // Luminu start
+    private java.util.Locale locale;
+
+    @Override
+    public java.util.Locale getSetLocale() {
+        return locale;
+    }
+
+    @Override
+    public void setLocale(java.util.Locale locale) {
+        this.locale = locale;
+    }
+    // Luminu end
+
     // Paper start
     public void setAffectsSpawning(boolean affects) {
         this.getHandle().affectsSpawning = affects;
