From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <28825609+KennyTV@users.noreply.github.com>
Date: Sat, 9 Nov 2019 13:24:50 +0100
Subject: [PATCH] Opt to disable data and log saving


diff --git a/src/main/java/net/minecraft/server/DedicatedServer.java b/src/main/java/net/minecraft/server/DedicatedServer.java
index 4c561181a977fd0244325880bb6a8cd6a54dcacc..743b36756a631199290cd1e928f863783e48b954 100644
--- a/src/main/java/net/minecraft/server/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/DedicatedServer.java
@@ -131,6 +131,14 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
         */
         // Paper end
 
+        // Luminu start - option to disable log file saving
+        java.util.Map<String, org.apache.logging.log4j.core.Appender> appenders = ((org.apache.logging.log4j.core.Logger) LogManager.getRootLogger()).getAppenders();
+        if (!this.propertyManager.getProperties().saveLogsToFile) {
+            appenders.remove("File");
+            logger.info("Disabled log file saving!");
+        }
+        // Luminu end
+
         // Paper start - Use Log4j IOStreams
         System.setOut(org.apache.logging.log4j.io.IoBuilder.forLogger(logger).setLevel(Level.INFO).buildPrintStream());
         System.setErr(org.apache.logging.log4j.io.IoBuilder.forLogger(logger).setLevel(Level.WARN).buildPrintStream());
diff --git a/src/main/java/net/minecraft/server/DedicatedServerProperties.java b/src/main/java/net/minecraft/server/DedicatedServerProperties.java
index 410d2595258cbf13900877ca4dd9d2872cb85eb0..18d27b8d4e8827ecf60ad8932896aba611b8d8bf 100644
--- a/src/main/java/net/minecraft/server/DedicatedServerProperties.java
+++ b/src/main/java/net/minecraft/server/DedicatedServerProperties.java
@@ -55,6 +55,8 @@ public class DedicatedServerProperties extends PropertyManager<DedicatedServerPr
     public final PropertyManager<DedicatedServerProperties>.EditableProperty<Boolean> whiteList;
     public final GeneratorSettings generatorSettings;
     public String serverName; // Luminu - readd (configurable) servername
+    public boolean saveDataOnStop; // Luminu - option to disable world and player data saving on stop
+    public boolean saveLogsToFile; // Luminu - option to disable log file saving
 
     public final String rconIp; // Paper - Add rcon ip
 
@@ -113,6 +115,8 @@ public class DedicatedServerProperties extends PropertyManager<DedicatedServerPr
         this.rconIp = rconIp == null ? this.serverIp : rconIp;
         // Paper end
         this.serverName = this.getString("server-name", "Unknown server"); // Luminu - readd servername
+        this.saveDataOnStop = this.getBoolean("save-world-on-stop", true); // Luminu - option to disable world and player data saving on stop
+        this.saveLogsToFile = this.getBoolean("save-log-files", true); // Luminu - option to disable log file saving
     }
 
     // CraftBukkit start
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index e1d8a0e885cc9d7270c973cf947b4ab09f9e200b..dec5a1e19f04603eec492c31ac23485689fe0970 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -780,6 +780,8 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
             this.getServerConnection().b();
         }
 
+        // Luminu start - add option to disable saving
+        if (server.getServer().propertyManager.getProperties().saveDataOnStop) { // don't indent because of larger diff :(
         if (this.playerList != null) {
             MinecraftServer.LOGGER.info("Saving players");
             this.playerList.savePlayers();
@@ -833,6 +835,7 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
             this.getUserCache().c(false); // Paper
         }
         // Spigot end
+        } // Luminu end
         // Paper start - move final shutdown items here
         LOGGER.info("Flushing Chunk IO");
         com.destroystokyo.paper.io.PaperFileIOThread.Holder.INSTANCE.close(true, true); // Paper
@@ -1033,13 +1036,17 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
                 crashreport = this.b(new CrashReport("Exception in server tick loop", throwable));
             }
 
+            // Luminu start - option to disable log file saving
+            if (server.getServer().propertyManager.getProperties().saveLogsToFile) {
             File file = new File(new File(this.A(), "crash-reports"), "crash-" + (new SimpleDateFormat("yyyy-MM-dd_HH.mm.ss")).format(new Date()) + "-server.txt");
-
             if (crashreport.a(file)) {
                 MinecraftServer.LOGGER.error("This crash report has been saved to: {}", file.getAbsolutePath());
             } else {
                 MinecraftServer.LOGGER.error("We were unable to save this crash report to disk.");
             }
+            }
+            MinecraftServer.LOGGER.error(crashreport.e()); // Luminu - log crash report
+            // Luminu end
 
             this.a(crashreport);
         } finally {
