From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <28825609+KennyTV@users.noreply.github.com>
Date: Sun, 15 Dec 2019 13:36:05 +0100
Subject: [PATCH] Player database id


diff --git a/src/main/java/net/minecraft/server/PlayerList.java b/src/main/java/net/minecraft/server/PlayerList.java
index 9382e8f79e8edec8885c629a36e230fbec50e1fb..4dc687f90bffc63d8938f73a531ad3f2398ed494 100644
--- a/src/main/java/net/minecraft/server/PlayerList.java
+++ b/src/main/java/net/minecraft/server/PlayerList.java
@@ -76,6 +76,7 @@ public abstract class PlayerList {
     // CraftBukkit start
     private CraftServer cserver;
     private final Map<String,EntityPlayer> playersByName = new java.util.HashMap<>();
+    public final it.unimi.dsi.fastutil.ints.Int2ObjectMap<EntityPlayer> playersById = new it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap<>(); // Luminu - database id
     @Nullable String collideRuleTeamName; // Paper - Team name used for collideRule
 
     public PlayerList(MinecraftServer minecraftserver, IRegistryCustom.Dimension iregistrycustom_dimension, WorldNBTStorage worldnbtstorage, int i) {
@@ -259,6 +260,7 @@ public abstract class PlayerList {
         playerconnection.a(entityplayer.locX(), entityplayer.locY(), entityplayer.locZ(), entityplayer.yaw, entityplayer.pitch);
         this.players.add(entityplayer);
         this.playersByName.put(entityplayer.getName().toLowerCase(java.util.Locale.ROOT), entityplayer); // Spigot
+        this.playersById.put(entityplayer.getBukkitEntity().getDatabaseId(), entityplayer); // Luminu - database id
         this.j.put(entityplayer.getUniqueID(), entityplayer);
         // this.sendAll(new PacketPlayOutPlayerInfo(PacketPlayOutPlayerInfo.EnumPlayerInfoAction.ADD_PLAYER, new EntityPlayer[]{entityplayer})); // CraftBukkit - replaced with loop below
 
@@ -548,6 +550,7 @@ public abstract class PlayerList {
         entityplayer.getAdvancementData().a();
         this.players.remove(entityplayer);
         this.playersByName.remove(entityplayer.getName().toLowerCase(java.util.Locale.ROOT)); // Spigot
+        this.playersById.remove(entityplayer.getBukkitEntity().getDatabaseId()); // Luminu - database id
         this.server.getBossBattleCustomData().b(entityplayer);
         UUID uuid = entityplayer.getUniqueID();
         EntityPlayer entityplayer1 = (EntityPlayer) this.j.get(uuid);
@@ -719,6 +722,7 @@ public abstract class PlayerList {
         entityplayer.stopRiding(); // CraftBukkit
         this.players.remove(entityplayer);
         this.playersByName.remove(entityplayer.getName().toLowerCase(java.util.Locale.ROOT)); // Spigot
+        this.playersById.remove(entityplayer.getBukkitEntity().getDatabaseId()); // Luminu - database id
         entityplayer.getWorldServer().removePlayer(entityplayer);
         BlockPosition blockposition = entityplayer.getSpawn();
         boolean flag1 = entityplayer.isSpawnForced();
@@ -849,6 +853,7 @@ public abstract class PlayerList {
             worldserver1.addPlayerRespawn(entityplayer1);
             this.players.add(entityplayer1);
             this.playersByName.put(entityplayer1.getName().toLowerCase(java.util.Locale.ROOT), entityplayer1); // Spigot
+            this.playersById.put(entityplayer.getBukkitEntity().getDatabaseId(), entityplayer); // Luminu - database id
             this.j.put(entityplayer1.getUniqueID(), entityplayer1);
         }
         // entityplayer1.syncInventory();
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 174c8215d507ab7dc09637545354dc1b67729050..4e2ab148a3ecbb24bbab9049c76e57d6e5e89bf3 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -520,6 +520,12 @@ public final class CraftServer implements Server {
     public net.luminu.paper.locale.PaperLocaleProvider getLocaleProvider() {
         return localeProvider;
     }
+
+    @Override
+    public Player getPlayer(int databaseId) {
+        EntityPlayer player = playerList.playersById.get(databaseId);
+        return (player != null) ? player.getBukkitEntity() : null;
+    }
     // Luminu end
 
     @Override
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 723996a703efea72d3dbf7e84a0555891b3106a8..84fb9154a267f8c1585ccfa14793667363565775 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1969,6 +1969,17 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
     // Luminu start
     private java.util.Locale locale = java.util.Locale.GERMAN;
+    private int databaseId;
+
+    @Override
+    public int getDatabaseId() {
+        return databaseId;
+    }
+
+    @Override
+    public void setDatabaseId(int databaseId) {
+        this.databaseId = databaseId;
+    }
 
     @Override
     public java.util.Locale getSetLocale() {
