From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KennyTV <28825609+KennyTV@users.noreply.github.com>
Date: Sun, 15 Dec 2019 13:36:05 +0100
Subject: [PATCH] Player database id


diff --git a/src/main/java/net/minecraft/server/PlayerList.java b/src/main/java/net/minecraft/server/PlayerList.java
index 787e80c50a61d6bd80efd8bd0d4b5140908aaeed..00c4f4c97e6284cdd5462df2e20990b655adee77 100644
--- a/src/main/java/net/minecraft/server/PlayerList.java
+++ b/src/main/java/net/minecraft/server/PlayerList.java
@@ -22,6 +22,9 @@ import javax.annotation.Nullable;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap; // Luminu
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap; // Luminu
+
 // CraftBukkit start
 import com.google.common.base.Predicate;
 import com.google.common.collect.Iterables;
@@ -76,6 +79,7 @@ public abstract class PlayerList {
     // CraftBukkit start
     private CraftServer cserver;
     private final Map<String,EntityPlayer> playersByName = new java.util.HashMap<>();
+    public final Int2ObjectMap<EntityPlayer> playersById = new Int2ObjectOpenHashMap<>(); // Luminu - database id
     @Nullable String collideRuleTeamName; // Paper - Team name used for collideRule
 
     public PlayerList(MinecraftServer minecraftserver, IRegistryCustom.Dimension iregistrycustom_dimension, WorldNBTStorage worldnbtstorage, int i) {
@@ -259,6 +263,7 @@ public abstract class PlayerList {
         playerconnection.a(entityplayer.locX(), entityplayer.locY(), entityplayer.locZ(), entityplayer.yaw, entityplayer.pitch);
         this.players.add(entityplayer);
         this.playersByName.put(entityplayer.getName().toLowerCase(java.util.Locale.ROOT), entityplayer); // Spigot
+        this.playersById.put(entityplayer.getBukkitEntity().getDatabaseId(), entityplayer); // Luminu - database id
         this.j.put(entityplayer.getUniqueID(), entityplayer);
         // this.sendAll(new PacketPlayOutPlayerInfo(PacketPlayOutPlayerInfo.EnumPlayerInfoAction.ADD_PLAYER, new EntityPlayer[]{entityplayer})); // CraftBukkit - replaced with loop below
 
@@ -549,6 +554,7 @@ public abstract class PlayerList {
         entityplayer.getAdvancementData().a();
         this.players.remove(entityplayer);
         this.playersByName.remove(entityplayer.getName().toLowerCase(java.util.Locale.ROOT)); // Spigot
+        this.playersById.remove(entityplayer.getBukkitEntity().getDatabaseId()); // Luminu - database id
         this.server.getBossBattleCustomData().b(entityplayer);
         UUID uuid = entityplayer.getUniqueID();
         EntityPlayer entityplayer1 = (EntityPlayer) this.j.get(uuid);
@@ -720,6 +726,7 @@ public abstract class PlayerList {
         entityplayer.stopRiding(); // CraftBukkit
         this.players.remove(entityplayer);
         this.playersByName.remove(entityplayer.getName().toLowerCase(java.util.Locale.ROOT)); // Spigot
+        this.playersById.remove(entityplayer.getBukkitEntity().getDatabaseId()); // Luminu - database id
         entityplayer.getWorldServer().removePlayer(entityplayer);
         BlockPosition blockposition = entityplayer.getSpawn();
         float f = entityplayer.getSpawnAngle();
@@ -861,6 +868,7 @@ public abstract class PlayerList {
             worldserver1.addPlayerRespawn(entityplayer1);
             this.players.add(entityplayer1);
             this.playersByName.put(entityplayer1.getName().toLowerCase(java.util.Locale.ROOT), entityplayer1); // Spigot
+            this.playersById.put(entityplayer.getBukkitEntity().getDatabaseId(), entityplayer); // Luminu - database id
             this.j.put(entityplayer1.getUniqueID(), entityplayer1);
         }
         // entityplayer1.syncInventory();
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index ff9f3f3e052ed2d3839212c215745ada241f083e..2148dc7656b052e14c9e06385688637ce2485c76 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -521,6 +521,12 @@ public final class CraftServer implements Server {
     public net.luminu.paper.locale.PaperLocaleProvider getLocaleProvider() {
         return localeProvider;
     }
+
+    @Override
+    public Player getPlayer(int databaseId) {
+        EntityPlayer player = playerList.playersById.get(databaseId);
+        return (player != null) ? player.getBukkitEntity() : null;
+    }
     // Luminu end
 
     @Override
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index d1d8f8ddf80a50777f37b6f12f5946e965d9e3f3..4e711ef86bdb89ec2887e23c6fd70b314961d0d9 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1962,6 +1962,17 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
     // Luminu start
     private java.util.Locale locale = java.util.Locale.GERMAN;
+    private int databaseId;
+
+    @Override
+    public int getDatabaseId() {
+        return databaseId;
+    }
+
+    @Override
+    public void setDatabaseId(int databaseId) {
+        this.databaseId = databaseId;
+    }
 
     @Override
     public java.util.Locale getSetLocale() {
